awx/main/tasks.py

```diff
@@ -904,6 +904,16 @@ def build_execution_environment_params(self, instance):
             "container_options": ['--user=root'],
         }
 
+        if instance.execution_environment.credential:
+            with open('/tmp/auth.json', 'w') as authfile:
+                image = instance.execution_environment.image
+                credentials = instance.execution_environment.credential.inputs
```

Comment:
`.inputs` contains encrypted values:

```python
In [2]: cred = Credential.objects.get(pk=4)

In [3]: cred.inputs
Out[3]:
{'host': 'https://quay.io/foo/bar',
 'password': '$encrypted$UTF8$AESCBC$Z0FBQUFBQmdXMmV3YkhuSjVYajhLdzRDOENpZDZZb2o5S2lvc2JkR2VwR1hrTl9ub3ZlajhhZkx2RXRyVDhGbHJ4N2c3eE9QN1MwSjFlNnpuWUNGYi1hMmQySklMSkJOSnc9PQ==',
 'username': 'ryan'}
```

You might want to explicitly build a dictionary and call `.get_input()`, which will perform the deencryption for the password:

```python
In [4]: {
   ...:     'username': cred.inputs['username'],
   ...:     'password': cred.get_input('password')
   ...: }
Out[4]: {'password': 'testing', 'username': 'ryan'}
```

Reply:
Or something like this, which won't populate keys that don't have values:

```In [8]: dict((k, cred.get_input(k)) for k in cred.inputs if cred.get_input(k) and k != 'host')
Out[8]: {'password': 'testing', 'username': 'ryan'}```

Reply:
better now?